src/App.tsx

```diff
@@ -5,21 +5,37 @@ import AddNewPlant from "./pages/addNewPlantPage/AddNewPlant.tsx";
 import GlobalStyle from "./theme/globalStyles.ts";
 import { theme } from "./theme/theme.ts";
 import { ThemeProvider } from "styled-components";
+import { FormValues } from "./components/form/Form.types.ts";
+
+const App = () => {
+  const token = localStorage.getItem("formData");
```

Comment:
Here and in other components that render on page load, I had to add validation. Using just  `const plantsCollection: FormValues[] = JSON.parse(localStorage.getItem("formData") || "") || [];` was throwing JSON parsing error on page load when local storage was empty (it couldn't parse null)

Reply:
it ended up a bit too verbose with all those || and double localStorage access. 

what would you say to this?
```
  const formData = localStorage.getItem("formData");

  const plantsCollection =
    formData ? JSON.parse(formData) as FormValues[]: [];
```

We access the storage once, then only if it's not nullable we parse it so that `JSON.parse()` can only fire if provided with meaningful data from the storage. Otherwise, we simply return an empty array that you can iterate over